### PR TITLE
snippets: emulate the nRF54L05 and the nRF54L10 RAM and RRAM sizes

### PR DESCRIPTION
**Pull request by carlescufi**
Note that this is *not* compatible with enabling FLPR.

### CODE REVIEW

**Review by gmarull — APPROVED** [review]

**Review by ihansse — APPROVED** [review]

File: snippets/emulated-nrf54l05/README.rst [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+Overview
+********
+
+Simulate the nRF54L05 on an nRF54L15.
+
+Supported SoCs and boards
+*************************
+
+Currently, the only SoC and board supported for use with the snippet is:
```
Comment (masz-nordic) [inline snippets/emulated-nrf54l05/README.rst:18]: It should be pointed out that this snippet is not compatible with FLPR snippets and target.
Same with L10.

Comment (carlescufi) [reply snippets/emulated-nrf54l05/README.rst:18]: Yep, right now it's only in the commit message but you are right.

**Comment by carlescufi** [comment]
@masz-nordic addressed your comment.

**Review by masz-nordic — APPROVED** [review]

**Comment by carlescufi** [comment]
@annwoj could you please review this one?

**Review by bjorn-tore-taraldsen — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

File: snippets/emulated-nrf54l10/snippet.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+name: emulated-nrf54l10
+
+boards:
+  nrf54l15dk/nrf54l15/cpuapp:
```
Comment (nordicjm) [inline snippets/emulated-nrf54l10/snippet.yml:10]: can use a wildcard for this so it works on any board with an nrf54l15 but not required

**Comment by divipillai** [comment]
> A snippet section will be added under nRF54l15 developing with guide in a separate PR. Also , added some suggestions of Anna, which was discussed offline.

Created PR https://github.com/nrfconnect/sdk-nrf/pull/18090 for the doc changes.

**Review by divipillai — APPROVED** [review]

**Review by wbober — APPROVED** [review]

**Comment by shanthanordic** [comment]
Manually tested for the following examples 
hello_world
bluetooth_peripheral_uart 
bluetooth_peripheral_lbs 
bluetooth_peripheral_hids_mouse 
peripheral_power_profiling

**Comment by carlescufi** [comment]
> @carlescufi Why this has milestone 2.8.0? When it was decided to add it to the scope?

Check with @shanthanordic please.

**Pull request merged**